### ISSUE TITLE
Duplicate / route when running sonata:page:update-core-routes command

### ISSUE THREAD

**Issue by sebastianblum** [sonata-project/SonataPageBundle#590]
Hello!

I work with PageBundle on 2.4.x-dev and first created a new Site and afterwards I run the command 

```
app/console sonata:page:update-core-routes --site=1
```

Now there are 2 pages as homepage created.
The first one with route_name page_slug and path /
the second one with route_name homepage and path /

If I try to edit one of the two homepages, I will get an error, that / already exists.

I found this solution: Delete the second one in the database and change the route_name to homepage. 

This is the part, which is wrong imo.
https://github.com/sonata-project/SonataPageBundle/blob/master/Route/RoutePageGenerator.php#L90

Can someone reproduce the problem?

greetings from munich,
Sebastian

**Comment by OskarStark** [comment]
@core23 can you reproduce this?

**Comment by core23** [comment]
@sebastianblum Can you give some more information how you created two pages with the same route? Do you have one static routing in the ``routing.yml`` file and one dynamic SonataPageBundle page route?

**Comment by sebastianblum** [comment]
@core23 
yes

First, I drop the database and create a new one.

    app/console sonata:page:create-site

    Please define a value for Site.name : Test
    Please define a value for Site.host : dev.sblum.de
    Please define a value for Site.relativePath : 
    Please define a value for Site.relativePath : /
    Please define a value for Site.enabled : true
    Please define a value for Site.enabledFrom : now
    Please define a value for Site.enabledTo : +10years
    Please define a value for Site.default : true
    Please define a value for Site.locale : de
    
    Creating website with the following information :
      name : +Test
      site : http(s)://dev.sblum.de
      enabled :  from Fri, 18 Dec 2015 12:12:19 +0100 => to Thu, 18 Dec 2025 12:12:19 +0100
    
    Confirm site creation ?y
    
    Site created !

Now second step:

    app/console sonata:page:update-core-routes --site=1

    ==================================================================================================
    
     > Updating core routes for site : +Test - http://dev.sblum.de
    
    ==================================================================================================
      CREATE  homepage                                           /
      CREATE  sonata_page_exceptions_list                        /exceptions/list
      CREATE  sonata_page_exceptions_edit                        /exceptions/edit/{code}
      CREATE  sonata_page_cache_esi                              /sonata/page/cache/esi/{_token}/{manager}/{page_id}/{block_id}
      CREATE  sonata_page_cache_ssi                              /sonata/page/cache/ssi/{_token}/{manager}/{page_id}/{block_id}
      CREATE  _page_internal_error_not_found                     
      CREATE  _page_internal_error_fatal                         
    
    done!


In the Admin, I get now 2 created homepages:

![bildschirmfoto 2015-12-18 um 12 16 42](https://cloud.githubusercontent.com/assets/551751/11895742/4f4bc2d0-a581-11e5-9202-1328027c7e2d.png)

This is the database:

![bildschirmfoto 2015-12-18 um 12 18 41](https://cloud.githubusercontent.com/assets/551751/11895774/89788ae2-a581-11e5-8ed2-aa3186b86b31.png)

Here are 2 screenshots, if I want to edit both homepages:

![bildschirmfoto 2015-12-18 um 12 17 52](https://cloud.githubusercontent.com/assets/551751/11895786/9b67743e-a581-11e5-8b28-3ffb51d18ad4.png)
![bildschirmfoto 2015-12-18 um 12 18 05](https://cloud.githubusercontent.com/assets/551751/11895789/9fded25a-a581-11e5-95f2-aa0c60431cc6.png)


If I delete the second homepage, the command ( sonata:page:update-core-routes) always creates a second homepage until I change the route name of the first one from "page_slug" to "homepage"

Thank you very much for your help
Sebastian

**Comment by OskarStark** [comment]
Ping @rande

**Comment by Avtonom** [comment]
@OskarStark  ping. same problem

**Comment by Avtonom** [comment]
@sebastianblum how to fix the problem?

**Comment by grimpows** [comment]
hey guys, fixed the probleme with, i think, the best way to do it (mean without change any code in vendor)
here the step to do :
//app/config/config.yml
parameters:
    sonata.page.admin.site.class: Application\Sonata\PageBundle\Admin\SiteAdmin
    sonata.page.route.page.generator.class: Application\Sonata\PageBundle\Route\RoutePageGenerator

//Application\Sonata\PageBundle\Admin\SiteAdmin
copy content from Sonata\PageBundle\Admin\SiteAdmin
change use Sonata\PageBundle\Route\RoutePageGenerator; -> use Application\Sonata\PageBundle\Route\RoutePageGenerator;

//if not symfony throw error with argument X passed in Sonata\PageBundle\Admin\SiteAdmin, expected a instance of Sonata\PageBundle\Route\RoutePageGenerator and get a instance of Application\Sonata\PageBundle\Route\RoutePageGenerator

//Application\Sonata\PageBundle\Route\RoutePageGenerator
copie content from Sonata\PageBundle\Route\RoutePageGenerator 

use Sonata\PageBundle\Route\RoutePageGenerator as BaseRoutePageGenerator;

class RoutePageGenerator extends BaseRoutePageGenerator
{
// you only need to overide update function
with the same modification as https://github.com/sonata-project/SonataPageBundle/pull/545


}

**Comment by core23** [comment]
Can you provide a PR with that fix @grimpows  ?

**Comment by grimpows** [comment]
the author juanolon seem to already did a pr for, as i linked https://github.com/sonata-project/SonataPageBundle/pull/545 i'm a bit new on git, i use it for private repo, so i think i'm not the best well placed to make one, mebbe the PR linked could be validate ? i dont understand why that was not validated .... i just gived a good way to got the same result as the juanolon fix purpose without change vendor :)

**Comment by gremo** [comment]
Any update on this? In the meanwhile this can be easily solved adding these to the `ignore_routes` node:

```yml
- sonata_user_registration_check_email
- sonata_user_registration_confirm
- sonata_user_registration_confirmed
- sonata_user_registration_register
- sonata_user_resetting_check_email
- sonata_user_resetting_request
- sonata_user_resetting_reset
- sonata_user_resetting_send_email
- sonata_user_security_check
- sonata_user_security_login
- sonata_user_security_logout
```

**Comment by grimpows** [comment]
the problem wasnt how to ignore some route, but a dupliquated root page when you create controller with '/' route, sonata Page bundle always create a homepage with '/' route even if you already got your own '/' homepage, the fix purposed by  juanolon (#545) correctly solve the problem but was never validated, i just show all step to use the fix without modifing vendor

**Comment by gremo** [comment]
Any update on this?

I usually fix the problem after a fresh install this way by command line:

```php
php app/console sonata:page:update-core-routes --site=all
php app/console doctrine:query:sql "DELETE FROM page WHERE route_name = 'homepage' AND url = '/'"
php app/console doctrine:query:sql "UPDATE page SET route_name = 'homepage', decorate = 0 WHERE url = '/'"
```
